src/dataExplorer/context/results/childResults.tsx

```diff
@@ -32,7 +32,9 @@ const modifiersToApply = (viewOptions: ViewOptions): SqlQueryModifiers => {
 
   // 2. smoothing transformation next
   const shouldSmooth =
-    viewOptions.smoothing?.columns?.length > 0 && viewOptions.smoothing?.applied
+    viewOptions.smoothing?.columns?.length > 0 &&
+    viewOptions.smoothing?.applied &&
+    viewOptions.smoothing.columns.length
```

Comment:
Race condition found. Fixed.

Reply:
Can you explain why adding the line `viewOptions.smoothing.columns.length` fix the bug? Seems like `viewOptions.smoothing?.columns?.length > 0` is already checking the length, but not really?